### PR TITLE
fix: preserve task chat targets and repair invalid proposals

### PR DESCRIPTION
**Pull request by matthiasn**
Existing checklist edits and migrations could fail because legacy checklist data overwrote the journal item ID with null or a stale value. Task-chat context now preserves the real ID, and chat guidance supports migration to existing tasks and the reason required for user-toggled checklist changes. A migration to an existing task no longer inherits an unrelated follow-up’s group ID.

Action planning now requires JSON-only output and gets one isolated repair attempt after JSON or proposal validation fails. Invalid attempts cannot leave review items behind; cancellation and transport failures are not retried. Already-configured task languages are rejected during validation as well as in guidance, so a proposal cannot promise a change the shared handler would skip.

The live eval now covers all 15 exposed task tools with 36 fixed scenarios, including exact IDs/arguments, batch contents, relationship direction, time edits, follow-up dependencies and negative controls. A declared synthetic action overlay supplies the missing checklist/time/label state; the canonical penguin query corpus and its expectations remain unchanged.

### Validation

- **GLM-5.3-Flash:** expanded main baseline 29/36; three sequential final runs **36/36 each (108/108)**. Same frozen cases, fixture, report bundle, model and parameters.
- **Ordinary chat:** canonical local/follow-up/wider/absent/category-boundary cases **15/15** across three runs.
- **62 targeted tests pass across scoped runs**, plus the opt-in live test skips normally. Seven regression failures were reproduced before their fixes. Replacing strict grading with tool-name-only grading makes seven grader tests fail. Review added twelve execution-claim examples; regrading all 151 saved action responses with the stricter scorer preserves every result, including the final 108/108.
- Dart MCP analyzer: zero diagnostics. Formatter, diff check and `make knowledge_check`: clean.
- Final CI on `ed239c48fa`: **32 successful checks, 2 skipped; 100% patch coverage**, project coverage 99.38%. All review threads resolved. GitHub still requires code-owner approval.

| Request | Median preparation | Range, 3 runs |
|---|---:|---:|
| Add checklist item | 1.54 s | 1.40–1.66 s |
| Check existing item | 1.98 s | 1.92–2.16 s |
| Record time | 2.32 s | 2.29–3.86 s |
| Create follow-up and migrate item | 2.33 s | 2.05–2.47 s |

These are production-builder question-to-built-review timings, excluding publication/UI rendering. Live evals prepare proposals without accepting or executing them. Of 108 final cases, 107 used two completions; one copied an unavailable target ID from the question, was blocked by independent validation, and correctly refused after the single repair. Approval-path tests additionally exercise the real planner, persistence, confirmation service and context loader with simulated handler mutations: follow-up IDs resolve before validation, and a hidden created task blocks migration. The positive test fails when both ID-resolution paths are disabled. No latency-speedup claim is made from the single baseline sample. Raw responses and timing artifacts remain outside the repository. The three-run action matrix measured the first commit; the later grouping-metadata correction is covered by its regression and leaves model guidance unchanged.

This follows #4260 before release, so its existing changelog fragment covers the feature. The separate cross-device approval policy discussed in [that review](https://github.com/matthiasn/lotti/pull/4260#discussion_r3998166363) remains unresolved; these evals do not establish cross-device execution safety.

### CODE REVIEW

File: knowledge/features/agents/query-chat.md [hunk 1]
```diff
@@ -367,6 +367,13 @@ linked time entries, visible same-category task targets, eligible labels and
 the current timer when it belongs to the task. No neighbour raw entries are
 loaded. This supports checklist additions/updates, time recording and timer
 text, task fields, labels, follow-up tasks, relationships and migrations.
+Checklist targets use the journal metadata ID; nullable or stale legacy IDs
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply knowledge/features/agents/query-chat.md:370]: Updated generated.at to 2026-09-13T04:00:59Z in 74dba1e528. make knowledge_check passes with no OKF warnings or Mermaid failures.

File: test/features/ai/eval/support/query_action_eval.dart [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+                answer.proposedActions.indexOf(followUp) >=
+                    answer.proposedActions.indexOf(item))) {
+          errors.add('invalid_follow_up_dependency');
+        }
+      }
+    }
+    if (remaining.isNotEmpty) errors.add('unexpected_actions');
+    if (expected.isEmpty &&
+        RegExp(
+          r"\bI(?:'ve| have)? (?:added|created|recorded|updated|changed|marked|deleted)\b",
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/eval/support/query_action_eval.dart]: Expanded the matcher to cover set/link/archive/restore/remove/assign/migrate/rename/start/log/move/check and the remaining mutation verbs, including typographic apostrophes and optional adverbs. Added twelve claim examples; the regression fails with the old matcher. Replayed all 151 archived action responses through the stricter Dart scorer: every previous pass/fail result is preserved, including the final 108/108. Fixed in 74dba1e528, with analyzer-only string cleanup in 6d8fb0d196.

File: lib/features/agents/query/query_task_action_planner.dart [hunk 3]
```diff
@@ -165,17 +205,14 @@ class QueryTaskActionPlanner {
...
+            name == TaskAgentToolNames.migrateChecklistItems &&
+            args['targetTaskId'] == 'new-task';
         await builder.addBatchItem(
           toolName: name,
-          args:
-              name == TaskAgentToolNames.migrateChecklistItems &&
-                  args['targetTaskId'] == 'new-task'
-              ? {...args, 'targetTaskId': newTaskId}
-              : args,
+          args: migratesToNewTask ? {...args, 'targetTaskId': newTaskId} : args,
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/agents/query/query_task_action_planner.dart:213]: Verified against the full approval path: confirmItem resolves migration arguments before invoking the QueryChatActionService dispatcher, and successful creation also persists the actual ID into sibling migration items. The dispatcher then loads fresh context; FollowUpTaskHandler has linked the created task to the source, so the real visible target is available. Added tests in ed239c48fa exercising planner → persisted proposal → real confirmation service → real context loader (with simulated handler mutations). They prove the dispatcher receives the actual created-task ID and that making the new task private blocks migration. The positive test fails when both in-memory and persisted ID resolution are disabled. No generated/group-ID bypass is needed; live target validation stays intact.
